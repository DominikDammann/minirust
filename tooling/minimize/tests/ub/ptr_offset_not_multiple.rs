#![feature(ptr_sub_ptr)]

use std::ptr;

fn main() {
    let data = [1u16, 2, 3, 4, 5];
    unsafe {
        // Negative offset in `sub_ptr` is UB.
        let ptr = ptr::from_ref(&data[0]).cast::<u8>().add(1).cast::<u16>();
        ptr::from_ref(&data[4]).sub_ptr(ptr);
    }
}
