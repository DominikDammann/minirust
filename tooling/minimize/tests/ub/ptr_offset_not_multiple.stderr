fatal error: UB: non-zero remainder in exact division
